### PR TITLE
[#208] Make checkboxes disableable

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves :rage1: 

Required to copy-paste the implementation of `customCheckboxField` and refactor it.

## Checklist:

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/summoner/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [ ] All new and existing tests pass.
- [ ] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [ ] I've used the [`stylish-haskell` file](https://github.com/kowainik/summoner/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.
- [ ] I changed code related to the generated project.
  - [ ] I created a new project using `summoner` and I checked that this specific feature is working as expected.

### CODE REVIEW

File: src/Summoner/Tui.hs [hunk 1]
```diff
@@ -75,7 +77,7 @@ data SummonForm
 
 -- Creates the inout form from the given initial 'TreasureChest'.
 mkForm :: forall e . SummonKit -> Form SummonKit e SummonForm
-mkForm = setFormConcat arrangeColumns . newForm
+mkForm sk = setFormConcat arrangeColumns $ newForm
```
Comment (willbasky) [inline src/Summoner/Tui.hs]: may be `sKit` or `skit` will be better.

File: src/Summoner/Tui.hs [hunk 2]
```diff
@@ -173,6 +179,10 @@ app = App
         VtyEvent V.EvResize {}       -> continue s
```
Comment (willbasky) [inline src/Summoner/Tui.hs]: `s`?

File: src/Summoner/Tui.hs [hunk 3]
```diff
@@ -173,6 +179,10 @@ app = App
...
         -- TODO: should we cancel on Esc and apply on Ctrl+Enter?
         VtyEvent (V.EvKey V.KEsc []) -> halt s
+        MouseDown n _ _ _ -> case n of
+            GitHubEnable  -> handleFormEvent ev s >>= continue . mkForm . formState
+            GitHubDisable -> handleFormEvent ev s >>= continue . mkForm . formState
+            _             -> handleFormEvent ev s >>= continue
```
Comment (willbasky) [inline src/Summoner/Tui.hs]: `ev`?

File: src/Summoner/Tui.hs (hunk 2 above)
Comment (ChShersh) [reply src/Summoner/Tui.hs]: `s` has type `Form SummonKit e SummonForm`. Maybe `form` is better name...

File: src/Summoner/Tui.hs (hunk 3 above)
Comment (ChShersh) [reply src/Summoner/Tui.hs]: `ev` is for event. This probably will be refactored later.

File: src/Summoner/Tui.hs (hunk 2 above)
Comment (willbasky) [reply src/Summoner/Tui.hs]: `formState` or `formApp`?

File: src/Summoner/Tui.hs (hunk 3 above)
Comment (willbasky) [reply src/Summoner/Tui.hs]: `event` if nobody uses it/

**Pull request merged**